consensus/src/pipeline/virtual_processor/processor.rs

```diff
@@ -304,20 +308,41 @@ impl VirtualStateProcessor {
     }
 
     fn maybe_update_pruning_point_and_candidate(self: &Arc<Self>) {
-        // TODO: call this method from resolve_virtual
+        let virtual_sp = self.virtual_state_store.read().get().unwrap().ghostdag_data.selected_parent;
+        if virtual_sp == self.genesis_hash {
+            return;
+        }
+
+        let ghostdag_data = self.ghostdag_store.get_compact_data(virtual_sp).unwrap();
         let pruning_read_guard = self.pruning_store.upgradable_read();
         let current_pp = pruning_read_guard.pruning_point().unwrap();
+        let current_pp_bs = self.ghostdag_store.get_blue_score(current_pp).unwrap();
         let current_pp_candidate = pruning_read_guard.pruning_point_candidate().unwrap();
-        let virtual_gd = self.virtual_state_store.read().get().unwrap().ghostdag_data.to_compact();
         let (new_pruning_point, new_candidate) =
-            self.pruning_manager.next_pruning_point_and_candidate_by_block_hash(virtual_gd, None, current_pp_candidate, current_pp);
+            self.pruning_manager.next_pruning_point_and_candidate_by_block_hash(ghostdag_data, None, current_pp_candidate, current_pp);
 
         if new_pruning_point != current_pp {
+            let mut past_pruning_points_to_add = Vec::new();
+            for current in self.reachability_service.backward_chain_iterator(virtual_sp, current_pp, false) {
```

Comment:
Why not iterate forward from the prev pp to the new one and check for the pruning window condition? Perhaps we can change to this later. It makes more sense because that exact loop is needed anyhow for building the new pp utxo set

Reply:
Or have `next_pruning_point_and_candidate_by_block_hash` return a list